src/generators/typescript/TypeScriptRenderer.ts

```diff
@@ -96,16 +100,29 @@ ${lines.map(line => ` * ${line}`).join('\n')}
       content.push(rendererProperty);
     }
 
+    if (this.model.additionalProperties !== undefined) {
+      const propertyName = findPropertyNameForAdditionalProperties(this.model);
+      const additionalProperty = await this.runPropertyPreset(propertyName, this.model.additionalProperties, PropertyType.additionalProperty);
+      content.push(additionalProperty);
+    }
+
     return this.renderBlock(content);
   }
 
-  renderProperty(propertyName: string, property: CommonModel): string {
+  renderProperty(propertyName: string, property: CommonModel, type: PropertyType = PropertyType.property): string {
     const name = FormatHelpers.toCamelCase(propertyName);
-    const signature = this.renderTypeSignature(property, { isRequired: this.model.isRequired(propertyName) });
-    return `${name}${signature};`;
+    
+    if (type === PropertyType.property) {
+      const signature = this.renderTypeSignature(property, { isRequired: this.model.isRequired(propertyName) });
+      return `${name}${signature};`;
+    } else if (type === PropertyType.additionalProperty) {
+      const additionalPropertyType = this.renderType(property);
+      return `${name}?: Map<String, ${additionalPropertyType}>;`;
+    }
+    return '';
```

Comment:
Please use the `switch` statement.

Reply:
👍 Checked the other locations, don't feel like a switch statement there makes sense.